// react-bootstrap-sweetalert/dist/demo/src/Demo.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import React from 'react';
import SweetAlert from '../../components/SweetAlert';
import reactTools from 'react-tools';
import { Button, Row, Col, Alert } from 'react-bootstrap';
import { examples } from './examples';
// @ts-ignore
window.React = React;
// @ts-ignore
window.SweetAlert = SweetAlert;
var defaultTextAreaValue = "\n<SweetAlert\n  success\n  title=\"Woot!\"\n  onConfirm={this.hideAlert}\n>\n  I did it!\n</SweetAlert>\n".trim();
var Demo = /** @class */ (function (_super) {
    __extends(Demo, _super);
    function Demo() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = {
            alert: null,
            textareaValue: defaultTextAreaValue,
        };
        _this.renderHeader = function () {
            return (React.createElement("div", { className: "jumbotron" },
                React.createElement("h1", null, "SweetAlert for React/Bootstrap"),
                React.createElement("p", null, 'A beautiful replacement for JavaScript\'s "alert"'),
                React.createElement("p", { className: "small" },
                    React.createElement("span", null, "This is just a rewrite for React/Bootstrap of the original "),
                    React.createElement("a", { href: "http://tristanedwards.me/sweetalert", target: "_blank" }, "SweetAlert"),
                    "."),
                React.createElement("p", null,
                    React.createElement("a", { className: "btn btn-outline btn-lg", href: "https://github.com/djorg83/react-bootstrap-sweetalert" },
                        React.createElement("span", { className: "fa fa-github" }),
                        " View on GitHub")),
                React.createElement("iframe", { src: "http://ghbtns.com/github-btn.html?user=djorg83&repo=react-bootstrap-sweetalert&type=watch&count=true", className: "social-share" })));
        };
        _this.buttonExample = function (type) {
            switch (type) {
                case 'custom':
                    _this.setState({
                        alert: (React.createElement(SweetAlert, { custom: true, showCancel: true, confirmBtnText: "Yes", cancelBtnText: "No", confirmBtnBsStyle: "primary", cancelBtnBsStyle: "default", customIcon: "/demo/assets/thumbs-up.jpg", title: "Do you like thumbs?", onConfirm: _this.hideAlert, onCancel: _this.hideAlert }, "You will find they are up!"))
                    });
                    break;
                case 'input':
                    _this.setState({
                        alert: (React.createElement(SweetAlert, { input: true, showCancel: true, cancelBtnBsStyle: "default", title: "An input!", placeholder: "Write something", onConfirm: function (response) { return _this.onReceiveInput(response); }, onCancel: _this.hideAlert }, "Write something interesting:"))
                    });
                    break;
                default:
                    _this.setState({
                        alert: (React.createElement(SweetAlert, { showCancel: true, confirmBtnText: "Continue", confirmBtnBsStyle: type, type: type, title: "Are you sure?", onCancel: _this.hideAlert, onConfirm: _this.hideAlert }, "You will not be able to recover this imaginary file!"))
                    });
                    break;
            }
        };
        _this.hideAlert = function () { return _this.setState({ alert: null }); };
        _this.onConfirm = function () { return _this.hideAlert(); };
        _this.onCancel = function () { return _this.hideAlert(); };
        _this.deleteFile = function () {
            _this.setState({
                alert: (React.createElement(SweetAlert, { success: true, title: "Deleted!", onConfirm: _this.hideAlert }, "Your imaginary file has been deleted."))
            });
        };
        _this.cancelDelete = function () {
            _this.setState({
                alert: (React.createElement(SweetAlert, { danger: true, title: "Cancelled", onConfirm: _this.hideAlert }, "Your imaginary file is safe :)"))
            });
        };
        _this.onReceiveInput = function (value) {
            _this.setState({
                alert: (React.createElement(SweetAlert, { success: true, title: "Nice!", onConfirm: _this.hideAlert },
                    "You wrote: ",
                    value))
            });
        };
        _this.runExample = function (_a) {
            var snippet = _a.snippet;
            return _this.setState({ alert: _this.evalJsxSnippet(snippet) });
        };
        _this.jsxInputElement = null;
        _this.setJsxInputElement = function (element) {
            _this.jsxInputElement = element;
        };
        _this.evalJsxSnippet = function (jsx) {
            try {
                var code = reactTools.transform(jsx);
                return eval(code);
            }
            catch (e) {
                console.log(e);
                return (React.createElement(SweetAlert, { type: "danger", title: "Whoops!", onConfirm: function () { return _this.hideAlert(); } }, "That didn't work."));
            }
        };
        _this.runInputExample = function () {
            _this.setState({
                alert: _this.evalJsxSnippet(_this.jsxInputElement.value),
            });
        };
        return _this;
    }
    Demo.prototype.render = function () {
        var _this = this;
        return (React.createElement("div", null,
            this.renderHeader(),
            React.createElement("div", { className: "container" },
                React.createElement("h2", null, "Install via npm"),
                React.createElement("pre", { className: "text-center lead" }, "$ npm install react-bootstrap-sweetalert"),
                React.createElement("hr", { style: {
                        marginTop: 50,
                        marginBottom: 50,
                        borderTopColor: '#BBBBBB'
                    } }),
                React.createElement("h1", null, "Sandbox"),
                React.createElement(Alert, { bsStyle: "info" }, "Modify the code below to make a SweetAlert. Be sure to use {this.hideAlert} for onConfirm and onCancel."),
                React.createElement("div", null,
                    React.createElement(Row, null,
                        React.createElement(Col, { sm: 2, className: "text-center" },
                            React.createElement("p", null,
                                React.createElement(Button, { bsStyle: "primary", onClick: this.runInputExample }, "Run it!"))),
                        React.createElement(Col, { sm: 10 },
                            React.createElement("textarea", { spellCheck: false, ref: this.setJsxInputElement, defaultValue: this.state.textareaValue, style: {
                                    height: 300,
                                    width: '100%',
                                    display: 'block',
                                    padding: 9.5,
                                    margin: '0 0 10px',
                                    fontSize: 15,
                                    lineHeight: '1.6',
                                    color: '#B7EC80',
                                    wordBreak: 'break-all',
                                    wordWrap: 'normal',
                                    backgroundColor: '#424140',
                                    border: '1px solid #ccc',
                                    borderRadius: 4,
                                    overflow: 'auto',
                                    whiteSpace: 'pre',
                                    resize: 'none'
                                } })))),
                React.createElement("hr", { style: {
                        marginTop: 50,
                        marginBottom: 50,
                        borderTopColor: '#BBBBBB'
                    } }),
                React.createElement("h2", null, "Examples"),
                React.createElement("h4", null, "Different types"),
                React.createElement(Row, null,
                    React.createElement(Col, { sm: 12, className: "text-center" }, [
                        'Primary',
                        'Info',
                        'Success',
                        'Warning',
                        'Danger',
                        'Custom',
                        'Input',
                    ].map(function (label) { return (React.createElement(Button, { style: { marginRight: 8 }, bsStyle: label === 'Input' || label === 'Custom' ? 'default' : label.toLowerCase(), onClick: function () { return _this.buttonExample(label.toLowerCase()); } }, label)); }))),
                examples.map(function (example, index) { return (React.createElement("div", { key: 'example-' + index },
                    React.createElement("h4", null, example.title),
                    React.createElement(Row, null,
                        React.createElement(Col, { sm: 2, className: "text-center" },
                            React.createElement("p", null,
                                React.createElement(Button, { bsStyle: "primary", onClick: function () { return _this.runExample(example); } }, "Try It"))),
                        React.createElement(Col, { sm: 10 },
                            React.createElement("pre", null, example.snippet.trim()))))); })),
            React.createElement("footer", null,
                React.createElement("ul", { className: "links" },
                    React.createElement("li", null,
                        React.createElement("a", { href: "http://github.com/djorg83" },
                            React.createElement("i", { className: "fa fa-github" }),
                            " GitHub")))),
            this.state.alert));
    };
    return Demo;
}(React.Component));
export default Demo;
